### DISCUSSION TITLE
Run each single test in a worker of its own?

### DISCUSSION THREAD

**Discussion by ffMathy** [vitest-dev/vitest#6926] [Q&A]
Right now, you can run a separate worker for each test file. Is it possible to also run a worker in each test? I require ultimate isolation in a particular range of tests. Since they use Netflix's Polly.js for HTTP recording and replaying, I can't have more than 1 instance of Polly in-memory at a time.

**Comment by AriPerkkio** [comment]
Out-of-the-box, no. Isolation is file specific only. Workers are set up before Vitest even knows how many test cases a test file has.

However you can use Vitest's Node API's to do this, though this will lead into _running-vitest-inside-vitest_ case. We do this in our internal test cases quite much. https://vitest.dev/advanced/api.html#startvitest

So inside a `test/it`, call `startVitest` to run completely new Vitest process.
